src/reference/asciidoc/amqp.adoc

```diff
@@ -2464,6 +2470,52 @@ public Binding binding() {
 }
 ----
 
+[[anonymous-queue]]
+===== AnonymousQueue
+
+In general, when needing a uniquely-named, exclusive, auto-delete queue, it is recommended that the `AnonymousQueue` is
+used instead of broker-defined queue names (using `""` as a `Queue` name will cause the broker to generate the queue
+name).
+
+This is because:
+
+1. The queues are actually declared when the connection to the broker is established; this is long after the beans are
+created and wired together; beans using the queue need to know its name.
+In fact, the broker might not even be running when the app is started.
+2. If the connection to the broker is lost for some reason, the admin will re-declare the `AnonymousQueue` with the same
+name.
+If we used broker-declared queues, the queue name would change.
+
+Starting with _version 1.5.3_, you can control the format of the queue name used by `AnonymousQueue` s.
+
+By default, the queue name is the String representation of a `UUID`; for example:
+`07afcfe9-fe77-4983-8645-0061ec61a47a`.
+
+You can now provide an `AnonymousQueue.NamingStrategy` implementation in a constructor argument:
+
+[source, java]
+----
+@Bean
+public Queue anon1() {
+    return new AnonymousQueue(new AnonymousQueue.Base64UrlNamingStrategy());
+}
+
+@Bean
+public Queue anon2() {
+    return new AnonymousQueue(new AnonymousQueue.Base64UrlNamingStrategy("foo-"));
+}
+----
+
+The first will generate a queue name prefixed by `spring.gen-` followed by a base64 representation of the `UUID`, for
+example: `spring.gen-MRBv9sqISkuCiPfOYfpo4g`.
```

Comment:
I haven't been able to find the RabbitMQ algorithm, but looks like it is exactly the same style.